File: packages/typescript/tests/system/dragSlider.test.ts
import { DragSliderTool } from "alumnium";
import { describe } from "vitest";
import { baseIt } from "./helpers.ts";

describe("Drag Slider", () => {
  const it = baseIt.override("setup", async ({ setup, skip }) => {
    return async (options) => {
      const result = await setup(options);
      const { isMobile } = result;

      // Drag slider is not implemented in Appium yet
      if (isMobile) skip("Drag slider is not implemented in Appium yet");

      return result;
    };
  });

  it("sets slider value", async ({ expect, setup }) => {
    const { al, $ } = await setup({
      extraTools: [DragSliderTool],
    });

    await $.navigate("slider.html");

    await al.do("drag the slider to 70");
    await al.check("the slider value is 70", {
      assert: expect.assert,
    });

    await al.do("drag the slider to 33");
    await al.check("the slider value is 33", {
      assert: expect.assert,
    });
  });
});

File: packages/python/examples/pytest/drag_slider_test.py
from os import getenv

from pytest import mark

from alumnium.tools import DragSliderTool

driver_type = getenv("ALUMNIUM_DRIVER", "selenium")


@mark.xfail("appium" in driver_type, reason="Drag slider is not implemented in Appium yet")
def test_drag_slider(al_factory, navigate):
    al = al_factory(extra_tools=[DragSliderTool])

    navigate("slider.html")

    al.do("drag the slider to 70")
    al.check("the slider value is 70")

    al.do("drag the slider to 33")
    al.check("the slider value is 33")
